### ISSUE TITLE
Add method to summarize table column info

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#2999]
This is a feature request for Astropy table (related to, but separate from #2998).

`HDUList` has the `info` method that prints a summary of the HDUs in the list.
How about adding an `info` or `summary` method to `Table` that prints a summary of which columns it contains plus some basic stats for numerical columns plus info on the number of "good" (i.e. non-masked) elements.

As an illustration what I mean, here's what [stilts tpipe](http://www.star.bristol.ac.uk/~mbt/stilts/sun256/tpipe.html) does:
```
$ stilts tpipe ./gammapy/datasets/data/tev_catalogs/tevcat.fits.gz omode=stats
Total Rows: 173
+------------------+------------+------------+------------+-----------+------+
| column           | mean       | stdDev     | min        | max       | good |
+------------------+------------+------------+------------+-----------+------+
| canonical_name   |            |            |            |           | 173  |
| catalog_id       |            |            |            |           | 173  |
| catalog_id_name  |            |            |            |           | 173  |
| catalog_name     |            |            |            |           | 173  |
| coord_dec        | -4.2542686 | 39.261513  | -69.16583  | 72.98361  | 173  |
| coord_gal_lat    | 2.575876   | 28.594221  | -87.96714  | 83.41788  | 173  |
| coord_gal_lon    | 174.05411  | 121.33482  | 0.87189746 | 359.94485 | 173  |
| coord_ra         | 206.73235  | 93.13067   | 1.6083333  | 359.78925 | 173  |
| coord_type       | -95.566475 | 18.114359  | -99        | 0         | 173  |
| discoverer       | 7.2543354  | 3.168495   | 1          | 19        | 173  |
| discovery_date   |            |            |            |           | 173  |
| distance_value   | 73.530266  | 463.1952   | 0.00183    | 3900.0    | 126  |
| distance_type    |            |            |            |           | 124  |
| distance_kpc     | 372831.47  | 818103.4   | 0.102      | 6261886.0 | 126  |
| eth              | 1126.8209  | 4123.756   | 0.4        | 35000.0   | 143  |
| flux             | 0.1374593  | 0.26819348 | 0.0017     | 2.0       | 140  |
| id               | 163.36417  | 52.96711   | 74         | 254       | 173  |
| observatory_name |            |            |            |           | 173  |
| other_names      |            |            |            |           | 96   |
| owner            | -37.687862 | 48.745483  | -99        | 2         | 173  |
| size_x           | 0.30347693 | 0.4286603  | 0.02       | 2.6       | 65   |
| size_y           | 0.24755384 | 0.3756975  | 0.02       | 2.6       | 65   |
| source_type      | 14.433526  | 9.782161   | 1          | 37        | 173  |
| source_type_name |            |            |            |           | 173  |
| spec_idx         | 2.5542648  | 1.0718743  | -3.6       | 5.4       | 102  |
| src_rank         | -0.6820809 | 10.654184  | -99        | 1         | 173  |
| variability      | -45.11561  | 49.39926   | -99        | 2         | 173  |
+------------------+------------+------------+------------+-----------+------+
```

**Comment by SaraOgaz** [comment]
Is there still a need for this?  It was kind of hard to follow all the other linked (and indirectly linked) issues.  And looking at the docs, it's doesn't seem like ``table.info()`` is doing this at the moment?

**Comment by taldcroft** [comment]
@SaraOgaz - this is done now, closed by #3731.
